### PR TITLE
Bug 1853654 - Advanced settings TestRail matching

### PR DESCRIPTION
**Pull request by AndiAJ**
These changes are part of our data alignment between testRail and our automated UI tests.

For more information please see this [document](https://docs.google.com/document/d/1i9kftL_BiL3Lw-wrkCEi78T9ikp0H0B0vrKO2Zr4IyY/edit?usp=sharing)

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1853654

### CODE REVIEW

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/SettingsAdvancedTest.kt [hunk 1]
```diff
@@ -50,6 +50,7 @@ class SettingsAdvancedTest {
         mockWebServer.shutdown()
     }
 
+    // TestRail link: https://testrail.stage.mozaws.net/index.php?/cases/view/2092699
     // Walks through settings menu and sub-menus to ensure all items are present
     @Test
     fun settingsAdvancedItemsTest() {
```
Comment (sv-ohorvath) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/SettingsAdvancedTest.kt:56]: Maybe rename this as the manual test to "verifyAdvancedSettingsSectionItemsTest"

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/SettingsAddonsTest.kt [hunk 2]
```diff
@@ -112,6 +115,7 @@ class SettingsAddonsTest {
         }
     }
 
+    // TestRail link: https://testrail.stage.mozaws.net/index.php?/cases/view/561600
     // Installs uBlock add-on and checks that the app doesn't crash while loading pages with trackers
     @SmokeTest
     @Test
```
Comment (sv-ohorvath) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/SettingsAddonsTest.kt:121]: Rename the manual test to match this test?

**Review by sv-ohorvath — APPROVED** [review]

**Review by isabelrios — APPROVED** [review]
LGTM! Nothing else to add in addition to Oana's comment

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/SettingsAdvancedTest.kt (hunk 1 above)
Comment (AndiAJ) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/SettingsAdvancedTest.kt:56]: Sounds good !

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/SettingsAddonsTest.kt (hunk 2 above)
Comment (AndiAJ) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/SettingsAddonsTest.kt:121]: Sure

**Pull request merged**